phoenix-core/src/main/java/org/apache/phoenix/iterate/RoundRobinResultIterator.java

```diff
@@ -114,6 +114,7 @@ public Tuple next() throws SQLException {
                 index = (index + 1) % size;
             }
         }
+        close();
```

Comment:
When RRIterator#peek() (next result) is null, we close the iterator. But if we don't get any next result after looping through all RRIterators, we should close() here, that's the plan right? 
Although this looks good, curious if caller can implement the logic of calling RRResultIterator#close() if RRResultIterator#next() returns null. But if caller logic turns out to be more complicated, sure we should go this route I believe.

Reply:
It seems we already do have practice of closing iterators when next() exhausts all rows e.g `ConcatResultIterator`, `LimitingResultIterator`. I think we are good here.

Reply:
> When RRIterator#peek() (next result) is null, we close the iterator. But if we don't get any next result after looping through all RRIterators, we should close() here, that's the plan right?

Yes. When all iterators set numScannersCacheExhausted , and we're at the end, getIterators() will return and empty collection, as close() is skipped.

> Although this looks good, curious if caller can implement the logic of calling RRResultIterator#close() if RRResultIterator#next() returns null. But if caller logic turns out to be more complicated, sure we should go this route I believe.

This is is the pattern everywhere in the code. This is also less error-prone, and encapsulates the logic within the class, I see no reason to change this, either.